### ISSUE TITLE
Struggling to make "*Org Select*" to reuse window

### ISSUE THREAD

**Issue by vbuzin** [wasamasa/shackle#65]
I have the following config in my init file: 

``` lisp
(use-package shackle
  :defer 1
  :config
  (setq shackle-rules
        '((helpful-mode   :select t   :same t)
          ("*Org Select*" :select t :same t)))
  (shackle-mode t))
```

The `*Org Select*` buffer is used to display capture templates when you invoke `org-capture`. It seems that this window refuses to be controlled by shackle, the other rule works as expected.

**Comment by wasamasa** [comment]
I've debugged this a bit and discovered the following odd code in Org, see if you can spot what's wrong (or scroll to the end):

```elisp
(org-switch-to-buffer-other-window "*Org Select*")

(defun org-switch-to-buffer-other-window (&rest args)
  "Switch to buffer in a second window on the current frame.
In particular, do not allow pop-up frames.
Returns the newly created buffer."
  (org-no-popups
   (apply 'switch-to-buffer-other-window args)))

(defmacro org-no-popups (&rest body)
  "Suppress popup windows.
Let-bind some variables to nil around BODY to achieve the desired
effect, which variables to use depends on the Emacs version."
  (if (org-version-check "24.2.50" "" :predicate)
      `(let (pop-up-frames display-buffer-alist)
         ,@body)
    `(let (pop-up-frames special-display-buffer-names special-display-regexps special-display-function)
       ,@body)))
```

tl;dr: Org is being stupid and let-binds `display-buffer-alist` for the duration of this call in a not completely outdated Emacs, thereby preventing user customizations (which is what this package happens to do). Go yell at them, not me.
